docs/UserGuide.md

```diff
@@ -14,298 +14,396 @@ bring an application to keep you aware of your deadlines and not miss them.
+  - A command will not be executed if any of the provided ids are invalid (not in the task list).
+  - Commands will be executed as if the list of ids is provided in increasing order.
+  - Examples: `1 3 5`, `99`
+- `FILTERS` can be one or more of the following:
+  - Boolean filters
+    - Replace `1` with `0` to invert the filter. 
+    - `-done 1` filters completed tasks.
+    - `-overdue 1` filters overdue (incomplete and past deadline) tasks.
+    - `-rep 1` filters recurring (`REPEAT_TIMES` > 0) tasks.
+  - Non-boolean filters
+    - `-all` selects all tasks (this overrides all other filters).
+    - `-desc STRING` filters tasks with descriptions containing the given string.
+    - `-email EMAIL_ADDRESS` filters tasks with the given email address.
+    - `-before DEADLINE` filters tasks with deadlines before the given deadline.
+    - `-after DEADLINE` filters tasks with deadlines after the given deadline.
+    - `-tags LIST_OF_TAGS` filters tasks that have all the given tags.
+    - `-prio PRIORITY_LEVEL` filters tasks with the given priority level.
+  - Example: `-done 0 -desc next` selects all incomplete tasks with `next` in their description. 
+
+### Others
+
+- Trailing, leading or consecutive spaces will be removed from commands.
+- Certain commands that have no effect, such as deleting a non-existent attribute from a task or marking an already complete task, can still be executed anyway.
+
+## Features
 
 ### Adding a task `add`
 
-Adds a new task to your To-Do list.
+Adds a new task with the given attributes to your To-Do list.
 
-Format: `add DESCRIPTION [-due DEADLINE] [-email EMAIL_ADDRESS] [-tags LIST_OF_TAGS] [-rep REPEAT_DURATION] [-prio PRIORITY_LEVEL]`
+Format: `add DESCRIPTION [-due DEADLINE] [-email EMAIL_ADDRESS] [-tags LIST_OF_TAGS] [-rep REPEAT_TIMES] [-prio PRIORITY_LEVEL]`
 
-- The format for `DEADLINE` is `dd/mm/yyyy hh:mm` or `dd-mm-yyyy hh:mm`, and cannot be in the past.
-- `EMAIL_ADDRESS` must be of a valid email address format.
-- `LIST_OF_TAGS` can consist of multiple tags, separated by a space. A task cannot have multiple of the same tag. Tags will be sorted in lexicographic order.
-- `REPEAT_DURATION` must be an integer from 0 to 2147483647, and can only be used if a `DEADLINE` is also provided. A new task will be added every week starting from the `DEADLINE` for `REPEAT_DURATION` times.
-- `PRIORITY_LEVEL` must be either 1, 2, or 3 (1: `Low`, 2: `Medium`, 3: `High`).
+- Refer to the [Restrictions](#restrictions) section for more information about the parameters.
 
 Example of usage and output: 
 
-`add todo -due 23/09/3000 23:59` creates a task with the description `todo`, and the deadline `23 Sep 3000 23:59`.
+`add todo -due 23/09/2023 23:59 -rep 10` creates a task with the description `todo`, the deadline `23 Sep 3000 23:59`, which will produce a copy of itself after `30 Sep 3000 23:59`.
```

Comment:
The deadline behind is not updated to 2023

Reply:
Kinda amazing you spotted that